updater/lib/dependabot/updater/operations/create_security_update_pull_request.rb

```diff
@@ -0,0 +1,401 @@
+          Dependabot.logger.info(
+            "Dependabot can't find a published or compatible non-vulnerable " \
+            "version for #{checker.dependency.name}. " \
+            "The latest available version is #{checker.dependency.version}"
+          )
+
+          service.record_update_job_error(
+            error_type: "security_update_not_found",
+            error_details: {
+              "dependency-name": checker.dependency.name,
+              "dependency-version": checker.dependency.version
+            },
+            dependency: checker.dependency
+          )
+        end
+
+        def record_pull_request_exists_for_latest_version(checker)
+          service.record_update_job_error(
+            error_type: "pull_request_exists_for_latest_version",
+            error_details: {
+              "dependency-name": checker.dependency.name,
+              "dependency-version": checker.latest_version&.to_s
+            },
+            dependency: checker.dependency
+          )
+        end
+
+        def record_pull_request_exists_for_security_update(existing_pull_request)
+          updated_dependencies = existing_pull_request.map do |dep|
+            {
+              "dependency-name": dep.fetch("dependency-name"),
+              "dependency-version": dep.fetch("dependency-version", nil),
+              "dependency-removed": dep.fetch("dependency-removed", nil)
+            }.compact
+          end
+
+          service.record_update_job_error(
+            error_type: "pull_request_exists_for_security_update",
+            error_details: {
+              "updated-dependencies": updated_dependencies
+            }
+          )
+        end
+
+        def earliest_fixed_version_message(lowest_non_vulnerable_version)
+          if lowest_non_vulnerable_version
+            "The earliest fixed version is #{lowest_non_vulnerable_version}."
+          else
+            "Dependabot could not find a non-vulnerable version"
+          end
+        end
+
+        def security_update_not_possible_message(checker, latest_allowed_version,
+                                                 conflicting_dependencies)
+          if conflicting_dependencies.any?
+            dep_messages = conflicting_dependencies.map do |dep|
+              "  #{dep['explanation']}"
+            end.join("\n")
+
+            dependencies_pluralized =
+              conflicting_dependencies.count > 1 ? "dependencies" : "dependency"
+
+            "The latest possible version that can be installed is " \
+              "#{latest_allowed_version} because of the following " \
+              "conflicting #{dependencies_pluralized}:\n\n#{dep_messages}"
+          else
+            "The latest possible version of #{checker.dependency.name} that can " \
+              "be installed is #{latest_allowed_version}"
+          end
+        end
+        ### END: Security Update Helpers
+
+        def requirements_to_unlock(checker)
```

Comment:
We should be able to fully move this method to `UpdateChecker` after #5902 and not keep it duplicated for all operations.